### PR TITLE
feat: New role ExtendedUser

### PR DESCRIPTION
**Pull request by marcometz**
Can See all Resources like an admin, but can only edit his own

SVAK-8

Bildschirmfoto 2023-11-16 um 22 12 28

### CODE REVIEW

File: app/views/events/index.html.erb [hunk 1]
```diff
@@ -37,7 +37,8 @@
             <td align="center"><%= visibility_switch(event, "EventRecord") %></td>
           <% end %>
           <td align="right" class="text-nowrap">
-            <%= link_to "Bearbeiten", edit_event_path(event.id), class: "btn btn-sm btn-secondary" %>
+            <% edit_label = editable_by_user?(event) ? "Bearbeiten" : "Ansehen" %>
```
Comment (donni106) [inline app/views/events/index.html.erb:40]: `link_to_label`, because it is not only `edit`?

File: app/controllers/constructions_controller.rb [hunk 2]
```diff
@@ -14,6 +14,7 @@ def index
           title
           payload
           dataProvider {
+            id
```
Comment (donni106) [inline app/controllers/constructions_controller.rb:17]: there are a few of these `id` additions, but they are not used currently, or am i wrong? what about not requesting them, if not used?

File: app/views/events/index.html.erb (hunk 1 above)
Comment (marcometz) [reply app/views/events/index.html.erb:40]: but we are rendering the edit form and linking to the edit controller...

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -15,6 +15,12 @@ def editor?
     false
   end
 
+  def editable_by_user?(item)
```
[the remark this answers is not in the document]
Comment (marcometz) [reply app/helpers/application_helper.rb:18]: there was an error on sending the form if we load the id of the assigned dataprovider in the edit form and send it back to mainserver, so ... i don't know

[the remark this answers is not in the document]
Comment (donni106) [reply app/helpers/application_helper.rb:18]: ok :/

**Review by donni106 — APPROVED** [review]

File: app/controllers/constructions_controller.rb (hunk 2 above)
Comment (marcometz) [reply app/controllers/constructions_controller.rb]: removed

**Review by donni106 — APPROVED** [review]

**Pull request merged**